### ISSUE TITLE
Linear confused by only one building block and repeat unit

### ISSUE THREAD

**Issue by joshkamm** [lukasturcani/stk#288]
I found that I was getting an error similar to this:

```python
~/miniconda3/lib/python3.7/site-packages/stk/molecular/topology_graphs/polymer/linear/linear.py in __init__(self, building_blocks, repeating_unit, num_repeating_units, orientations, random_seed, reaction_factory, num_processes)
    251         self._num_repeating_units = num_repeating_units
    252 
--> 253         head, *body, tail = orientations
    254         choices = [True, False]
    255         vertices = [

ValueError: not enough values to unpack (expected at least 2, got 1)
```

After a bit of debugging I realized that I had multiple layers of construction and the error seemed to occur when one of the layers was functioning as a constant layer (constructing a linear polymer with only one monomer). I verified the hypothesis with this adaptation of an example from the docs which produced the error above:

```python
import stk

# React the amine functional groups during construction.
bb1 = stk.BuildingBlock('NCCN', [stk.PrimaryAminoFactory()])
# React the aldehyde functional groups during construction.
bb2 = stk.BuildingBlock('O=CCCC=O', [stk.AldehydeFactory()])
# Build a polymer.
polymer = stk.ConstructedMolecule(
    topology_graph=stk.polymer.Linear(
        building_blocks=(bb1, bb2),
        repeating_unit='A',
        num_repeating_units=1,
    ),
)
```

There is no error if `repeating_unit == 'AB'` or if `num_repeating_units == 2`. It seems like `Linear` isn't prepared for the edge case of `num_repeating_units == 1` and `repeating_unit` only having one building block. In basic usage it makes sense that you wouldn't bother constructing a molecule that's the same as one of your building blocks but it'd be convenient to not have to think about that as a special case when it occurs in multi-layered construction.

**Comment by lukasturcani** [comment]
Thanks for reporting this. Yeah, there is no real need for the edge case to exist, so  I agree it should be removed.

**Comment by lukasturcani** [comment]
This is now available in the new stk release, `v2021.5.26.0`,  available on pip, so existing stk installations can be updated with `pip install stk --upgrade`.
